lib/src/utils/loader.dart

```diff
@@ -1,303 +1,2 @@
-    final uri = Uri.parse(url);
-
-    final id = _TemporaryFileIdentifier(_Source.url, url);
-
-    // TODO(filiph): Add the option to check the filesystem first.
-    //               This could be a cache-invalidation problem (if the asset
-    //               changes from one version to another). But it could speed
-    //               up start up times.
-    if (_temporaryFiles.containsKey(id)) {
-      final existingFile = _temporaryFiles[id]!;
-      if (existingFile.existsSync()) {
-        _log.finest(
-          () => 'Sound from $url already exists as a temporary file.',
-        );
-        return _temporaryFiles[id]!;
-      }
-    }
-
-    final directory = _temporaryDirectory;
-
-    if (directory == null) {
-      throw const SoLoudTemporaryFolderFailedException(
-          "Temporary directory hasn't been initialized, "
-          'yet loadUrl() is called.');
-    }
-
-    final newFilepath = _getFullTempFilePath(id);
-    final newFile = File(newFilepath);
-
-    final Uint8List byteData;
-
-    try {
-      http.Response response;
-      if (httpClient != null) {
-        response = await httpClient.get(uri);
-      } else {
-        response = await http.get(uri);
-      }
-      if (response.statusCode == 200) {
-        byteData = response.bodyBytes;
-      } else {
-        throw SoLoudNetworkStatusCodeException(
-          response.statusCode,
-          'Failed to fetch file from URL: $url',
-        );
-      }
-    } catch (e) {
-      _log.severe(() => 'Error fetching $url', e);
-      rethrow;
-    }
-
-    final buffer = byteData.buffer;
-    try {
-      await newFile.create(recursive: true);
-      await newFile.writeAsBytes(
-        buffer.asUint8List(byteData.offsetInBytes, byteData.lengthInBytes),
-      );
-    } catch (e) {
-      throw SoLoudTemporaryFolderFailedException(
-        "loadAsset() couldn't write $newFile to disk",
-      );
-    }
-
-    _temporaryFiles[id] = newFile;
-
-    return newFile;
-  }
-
-  String _getFullTempFilePath(_TemporaryFileIdentifier id) {
-    final directory = _temporaryDirectory;
-
-    if (directory == null) {
-      throw StateError("Temporary directory hasn't been initialized, "
-          'yet _getTempFile() is called.');
-    }
-
-    return path.join(directory.absolute.path, id.asFilename);
-  }
-}
-
-enum _Source {
-  url,
-  asset,
-}
-
-@immutable
-class _TemporaryFileIdentifier {
-  const _TemporaryFileIdentifier(this.source, this.path);
-
-  final _Source source;
-
-  final String path;
-
-  String get asFilename =>
-      'temp-sound-${source.name}-0x${path.hashCode.toRadixString(16)}';
-
-  @override
-  int get hashCode => Object.hash(source, path);
-
-  @override
-  bool operator ==(Object other) {
-    return other is _TemporaryFileIdentifier &&
-        other.source == source &&
-        other.path == path;
-  }
-}
+export 'package:flutter_soloud/src/utils/loader_io.dart'
+    if (dart.library.html) 'package:flutter_soloud/src/utils/loader_web.dart';
```

Comment:
[nit] It would be a bit cleaner to have a stub implementation here, like discussed in https://dart.dev/guides/libraries/create-packages#conditionally-importing-and-exporting-library-files.

Something like

```dart
export 'loader_base.dart' // Stub implementation
    if (dart.library.io) 'loader_io.dart' // dart:io implementation
    if (dart.library.js_interop) 'loader_web.dart'; // package:web implementation
```

The stub implementation is an interface class, something like

```dart
interface class Loader {
  bool get blah => throw UnsupportedError('platform not supported');

  void doSomething() => throw UnsupportedError('platform not supported');
}

```

Reply:
This is not a big deal, though. Maybe just add a TODO comment with the link.